### PR TITLE
Defer notifications while performing action

### PR DESCRIPTION
**Pull request by ShirShintel**
- Add `deferSubscriberNotifications` Function that defers subscribers (and observables) notification until the end of some action. This is necessary when running a flow that creates intermediate states we don't want to react to. 

- To handle the conflict between `deferSubscriberNotifications` and `flush`, add optional config to the flush function:
   - `excecutionType: scheduled`- meaning flush would happen right after we stop deferring notifications
   - `excecutionType: immediate` - meaning flush would happen immediately regardless if notifications are currently being deferred.

In general, the `deferSubscriberNotifications` solution is:

- `deferSubscriberNotifications` receives some action we would like to perform while deferring notifications.
- If notifications are already deferred, run action without doing anything else (to support nesting `deferSubscriberNotifications`)
-  Set `deferNotifications` flag to true before running the action
-  When calling `notifyAll` while `deferNotifications` is set to true, return without notifying
-  Finally, set `deferNotifications` back to false, and flush\notify if someone tried to flush (not immediate) or notify while deferring notifications.

### CODE REVIEW

File: test/connectWithShell.spec.tsx [hunk 1]
```diff
@@ -636,6 +637,116 @@ describe('connectWithShell-useCases', () => {
...
+    it('should notify after action failed while deferring notifications', async () => {
+        const { host, shell, renderInShellContext } = createMocks(entryPointWithState, [entryPointSecondStateWithAPI])
+        const ConnectedComp = connectWithShell(mapStateToProps, undefined, shell, { allowOutOfEntryPoint: true })(PureComp)
+
+        const { testKit } = renderInShellContext(<ConnectedComp />)
+        if (!testKit) {
+            throw new Error('Connected component failed to render')
+        }
+
+        expect(testKit.root.findByProps({ className: 'ONE' }).children[0]).toBe('init1')
```
Comment (shirlynwix) [inline test/connectWithShell.spec.tsx:677]: Maybe you can extract `testKit.root.findByProps({ className: 'ONE' }).children[0]` to a function 🙃

**Review by shirlynwix — APPROVED** [review]
Very nice!
See small suggestions

File: test/connectWithShell.spec.tsx (hunk 1 above)
Comment (ShirShintel) [reply test/connectWithShell.spec.tsx:677]: Absolutely :)

File: src/throttledStore.tsx [hunk 2]
```diff
@@ -46,7 +46,8 @@ export interface StateContribution<TState = {}, TAction extends AnyAction = AnyA
 
 export interface ThrottledStore<T = any> extends Store<T> {
     hasPendingSubscribers(): boolean
-    flush(): void
+    flush(config?: { excecutionType: 'scheduled' | 'immediate' }): void
+    deferSubscriberNotifications<K>(action: () => K | Promise<K>): Promise<K>
```
[the remark this answers is not in the document]
Comment (ShirShintel) [reply src/throttledStore.tsx:50]: It talks directly with the store, so decided to keep it there

**Pull request merged**